src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -382,6 +388,10 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
     const {organization, ruleId, rule, params, onSubmitSuccess} = this.props;
     const {query, aggregation, timeWindow, triggers} = this.state;
 
+    const queryAndAlwaysErrorEvents = !query.includes('event.type')
```

Comment:
So we're only adding `!event.type:transaction` if they haven't filtered by event.type? Would this lead to undesirable behaviour if they filtered by `event.type:transaction`?

Reply:
I approved because I feel this is still a good short-term fix to the issue, and any bad edge cases don't outweigh the major case that's been fixed.

Reply:
+1

Reply:
Indeed - this will change with perf alerts, so I'm ok with the edgecases as well.